test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightBatchReaderTests.cs

```diff
@@ -363,6 +363,114 @@ public async Task ReadBatchRequestWithChangesetAndDependsOnIdsAsync()
                 isResponse: false);
         }
 
+        [Fact]
+        public async Task ReadBatchRequestWithNullHeadersAsync()
+        {
+            var payload = "{\"requests\": [{" +
+                "\"id\": \"1\"," +
+                "\"method\": \"POST\"," +
+                "\"url\": \"http://tempuri.org/Customers\"," +
+                "\"headers\": {\"odata-version\":\"4.0\",\"content-type\":\"application/json;odata.metadata=minimal;odata.streaming=true;IEEE754Compatible=false;charset=utf-8\",\"null-header\":null}, " +
+                "\"body\": {\"@odata.type\":\"#NS.Customer\",\"Id\":1,\"Name\":\"Customer 1\",\"Type\":\"Retail\"}}]}";
+
+            await SetupJsonLightBatchReaderAndRunTestAsync(

```

Comment:
@apiaskowski Not a deal breaker but if we're looking at this from a best practices point of view, the better way to assert on exceptions is this:

```csharp
var exception = await Assert.ThrowsAsync<ODataException>( // Or ArgumentException as the case maybe
    () => SetupJsonLightBatchReaderAndRunTestAsync(
        // ...
    );

Assert.Equal(/** expected error message **/, exception.Message);
```

Reply:
Sure, just this use-case confirms, that no Exception is thrown and not that an specific Exception is thrown and catched.
I am not sure if this is possible with the `Assert.ThrowsAsync` method, but I'll confirm it.

Reply:
You'll find tests within this test file that do it the way I described and verify that the specific exception that is thrown and even assert on the exception message...

Reply:
I've added the Assert.ThrowsAsync and Assert.Throws to the test cases of duplicate headers and properties.
Adding it to the null header test case does not work, since the Throws methods will cause the unit test to fail, if no exception is thrown, which the exact opposite to the test case purpose.